flake.nix

```
@@ -407,10 +407,18 @@
 
         # docker image with Nix inside
         dockerImage = nixpkgs.lib.genAttrs linux64BitSystems (system:
-          import ./docker.nix {
+          let
             pkgs = nixpkgsFor.${system};
-            tag = version;
-          });
+            image = import ./docker.nix { inherit pkgs; tag = version; };
+          in pkgs.runCommand "docker-image-tarball-${version}"
+            { meta.description = "Docker image with Nix for ${system}";
+            }
+            ''
+              mkdir -p $out/nix-support
+              image=$out/image.tar.gz
+              cp ${image} $image
+              echo "file binary-dist $image" >> $out/nix-support/hydra-build-products
```

```suggestion
              echo "file binary-dist ${image}" >> $out/nix-support/hydra-build-products
```

Copying is expensive (especially for large images) since it results in two copies in the binary cache.

---

Tnx, totally makes sense.

I did however did it a little differently, just so I could later use this hydraJob to push the image to docker hub.